packages/calcite-components/src/components/text-area/text-area.tsx

```diff
@@ -403,25 +403,37 @@ export class TextArea
   };
 
   renderCharacterLimit = (): VNode => {
-    return this.maxLength ? (
-      <span class={CSS.characterLimit}>
-        <span class={{ [CSS.characterOverLimit]: this.value?.length > this.maxLength }}>
-          {this.getLocalizedCharacterLength()}
+    if (this.maxLength) {
+      const { currentLength, maxLength } = this.getLocalizedCharactersLength();
+      return (
+        <span class={CSS.characterLimit}>
+          <span class={{ [CSS.characterOverLimit]: this.value?.length > this.maxLength }}>
+            {currentLength}
+          </span>
+          {"/"}
+          {maxLength}
         </span>
-        {"/"}
-        {numberStringFormatter.localize(this.maxLength.toString())}
-      </span>
-    ) : null;
+      );
+    }
   };
 
-  getLocalizedCharacterLength(): string {
-    numberStringFormatter.numberFormatOptions = {
-      locale: this.effectiveLocale,
-      numberingSystem: this.numberingSystem,
-      signDisplay: "never",
-      useGrouping: this.groupSeparator
-    };
-    return numberStringFormatter.localize(this.value ? this.value.length.toString() : "0");
+  getLocalizedCharactersLength(): { currentLength: string; maxLength: string } {
+    const currentLength = this.value ? this.value.length.toString() : "0";
+    const maxLength = this.maxLength.toString();
+    if (this.numberingSystem !== "latn") {
+      numberStringFormatter.numberFormatOptions = {
+        locale: this.effectiveLocale,
+        numberingSystem: this.numberingSystem,
+        signDisplay: "never",
+        useGrouping: this.groupSeparator
+      };
+      return {
+        currentLength: numberStringFormatter.localize(currentLength),
+        maxLength: numberStringFormatter.localize(maxLength)
+      };
+    } else {
```

Comment:
Some readability suggestions:

* Invert the if conditions so that the shorter block comes up first.
* Since you're already returning above, you can drop the `else`.

```ts
if (this.numberingSystem === "latn") {
  return { currentLength, maxLength };
}

numberStringFormatter.numberFormatOptions = {
  locale: this.effectiveLocale,  
  numberingSystem: this.numberingSystem,
  signDisplay: "never",
  useGrouping: this.groupSeparator
};

return {
  currentLength: numberStringFormatter.localize(currentLength),
  maxLength: numberStringFormatter.localize(maxLength)
};
```